src/GNDStk/tree/test/ctor.test.cpp

```diff
@@ -0,0 +1,184 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+
+/*
+What we'll need to test...
+
+   1. Tree()
+   2. Tree(Tree &&)
+
+   3. Tree(xml)
+   4. Tree(json)
+
+   5. Tree(char *)
+   6. Tree(string)
+   7. Tree(istream)
+
+   8. Tree(Tree   &) // copy
+   9. Tree(Tree<> &) // different <>
+*/
+
+
+
+// -----------------------------------------------------------------------------
+// ctor(from,ossf)
+// -----------------------------------------------------------------------------
+
+// TREE = a existing sosurce tree that arrives here as something
+//        we'll use to construct a new tree
+// M = metadata container class for the new tree
+// C = children container class for the new tree
+template<
+   class TREE,
+   template<class ...> class M = std::vector,
+   template<class ...> class C = std::vector
+>
+bool ctor(
+   const TREE &from, // arriving tree
+   const std::ostringstream &ossf // the arriving tree printed to a string
+) {
+   // construct
+   // The context here is such that we're testing a whole bunch of constructions
+   // of a Tree<A,B> from a Tree<C,D>, where in some cases <A,B> == <C,D>,
+   // in which case we're testing the copy constructor, and in other cases,
+   // <A,B> != <C,D>, in which case we're testing that our various tree species
+   // construct properly from other tree species.
+   gnds::Tree<M,C> to(from);
+
+   // print "to" to a string
+   std::ostringstream osst;
+   osst << to;
+
+   // compare the two strings
+   return osst.str() == ossf.str();
+}
+
+
+
+// -----------------------------------------------------------------------------
+// ctor()
+// -----------------------------------------------------------------------------
+
+// Tests various tree constructors, with an argument of Tree<M,C>
+// M = metadata container class
+// C = children container class
+template<
+   template<class ...> class M = std::vector,
+   template<class ...> class C = std::vector
+>
+bool ctor()
+{
+   // read a meaningful GNDS hierarchy into the tree we'll use as a source
+   const gnds::Tree<M,C> from("n-008_O_016.xml");
+
+   // print the tree to a string, which we'll soon use for comparison
+   std::ostringstream ossf;
+   ossf << from;
+
+   // construct from the tree into other tree variants, and check
+   // that the contents of each such result are the same
+   return
```

Comment:
Famous last words: "if it works for one, then it will work for the others." :-D

Reply:
Given that the assignment test (which I've reformulated, as I mentioned in another comment) tests lots of combinations, and given that the constructors and the assignments use the same code under-the-hood, it would probably be safe at this point to just test a handful of different combinations. Or, I could reformulate this test into something that may be clearer, as I did with the assignment test.

Reply:
Oops, probably be safe. I just spoke some more famous last words. :-p